File: Domain/Chapter.cs
namespace Domain
{
    public class Chapter : BaseEntity
    {
        public required string Title { get; set; }
        public string? Description { get; set; }
        public int Position { get; set; }
        public bool IsPublished { get; set; }
        public bool IsFree { get; set; }

        public Guid CourseId { get; set; }
        public required Course Course { get; set; }

        public ICollection<UserProgress> UserProgresses { get; set; } = [];

        public Video? Video { get; set; }
    }
}

File: angular-client-app/src/app/shared/types/chapter.ts
import { UserProgress } from '.';

export type Chapter = {
  id: string;
  title: string;
  position: number;
  description?: string;
  videoUrl?: string;
  videoContentType?: string;
  isPublished: boolean;
  isFree: boolean;
  userProgress: UserProgress[];
  isCompleted?: boolean;
  nextChapterId?: string;
};
